src/engine.rs

```diff
@@ -222,16 +222,16 @@ impl Engine {
             .unwrap_or_else(U256::zero)
     }
 
-    pub fn remove_storage(address: &Address, key: &H256) {
-        sdk::remove_storage(&storage_to_key(address, key));
+    pub fn remove_storage(address: &Address, key: &H256, generation: u32) {
```

Comment:
What do you think about creating a separate struct:

```rust
AddressWithGeneration {
    address: Address,
    generation: u64,
}
```

You can use a different (and better) name for it. 

This way interface for all this method will be a little bit cleaner.

Reply:
Typically I do stuff like that if we get over 5 arguments. Here, I didn't see much of a point but, I get that grouping it together would be fine.